concrete-csprng/src/counter/mod.rs

```diff
@@ -80,21 +84,27 @@ pub enum ShouldGenerateBatch {
 pub struct State {
     aes_ctr: AesCtr,
     byte_ctr: ByteCtr,
+    step_ctr: StepCtr,
 }
 
 impl State {
     /// Generate a new state from an aes counter and a byte counter.
     #[allow(dead_code)]
-    pub fn new(aes_ctr: AesCtr, byte_ctr: ByteCtr) -> Self {
+    pub fn new(aes_ctr: AesCtr, byte_ctr: ByteCtr, step_ctr: StepCtr) -> Self {
         debug_assert!(byte_ctr.0 <= 127);
-        State { aes_ctr, byte_ctr }
+        State {
+            aes_ctr,
+            byte_ctr,
+            step_ctr,
+        }
     }
 
     /// Generates a new state from an AES counter value.
     pub fn from_aes_counter(aes_ctr: AesCtr) -> Self {
         State {
             aes_ctr,
             byte_ctr: ByteCtr(0),
+            step_ctr: StepCtr(1),
```

Comment:
Why does it start at 1?

Reply:
It is the number of steps used when incrementing I believe.

Reply:
The step is the increment of the counter, when it moves to the next value. By default, the next value of the generator is the actual next value of the stream, so the step is 1.